homeassistant/components/sensor/google_travel_time.py

```diff
@@ -52,17 +52,17 @@
     vol.Required(CONF_DESTINATION): cv.string,
     vol.Required(CONF_ORIGIN): cv.string,
     vol.Optional(CONF_NAME): cv.string,
-    vol.Optional(CONF_TRAVEL_MODE): vol.In(TRAVEL_MODE),
+    vol.Optional(CONF_TRAVEL_MODE): cv.string,
     vol.Optional(CONF_OPTIONS, default={CONF_MODE: 'driving'}): vol.All(
         dict, vol.Schema({
-            vol.Optional(CONF_MODE, default='driving'): vol.In(TRAVEL_MODE),
+            vol.Optional(CONF_MODE, default='driving'): cv.string,
             vol.Optional('language'): vol.In(ALL_LANGUAGES),
             vol.Optional('avoid'): vol.In(AVOID),
             vol.Optional('units'): vol.In(UNITS),
             vol.Exclusive('arrival_time', 'time'): cv.string,
             vol.Exclusive('departure_time', 'time'): cv.string,
             vol.Optional('traffic_model'): vol.In(TRAVEL_MODEL),
-            vol.Optional('transit_mode'): vol.In(TRANSPORT_TYPE),
+            vol.Optional('transit_mode'): cv.string,
```

Comment:
This is a bad approach, we should never overload values to be either an entity or a value.

Reply:
I appreciate the feedback, but it's not very constructive: the quote doesn't seem to match the statement, and there is no indication of what an acceptable approach might be.

Are you suggesting using self._mode_entity_id for entity, and self._mode for value? Is that what would be a better approach? It means a lot more code, as this is all wrapped in a (largely unnecessary) self._options dict, but it is plausible. Maybe I should tidy that out too, and handle the options properly.